java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -83,48 +119,67 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
     }
 
     /**
-     * Sends the provided [ProxyMessage] to the store backing the provided [referenceId].
+     * Sends the provided [ProxyMessage] to the store backing the provided [muxId].
      *
-     * A new store will be created for the [referenceId], if necessary.
+     * A new store will be created for the [muxId], if necessary.
      */
     suspend fun onProxyMessage(
-        message: ProxyMessage<Data, Op, T>,
-        referenceId: String
+        message: ProxyMessage<Data, Op, T>
     ): Boolean {
-        val (id, store) = store(referenceId)
-        val deMuxedMessage: ProxyMessage<Data, Op, T> = when (message) {
-            is SyncRequest -> SyncRequest(id)
-            is ModelUpdate -> ModelUpdate(message.model, id)
-            is Operations -> if (message.operations.isNotEmpty()) {
-                Operations(message.operations, id)
-            } else return true
+        val muxId = requireNotNull(message.muxId) {
+            "messages sent to Direct Store Muxer must have a muxId"
         }
-        return store.onProxyMessage(deMuxedMessage)
+        val (idSet, store) = store(muxId)
+
+        if (!idSet.contains(message.id)) registerToStore(message.id!!, muxId)
```

Comment:
since you are doing the contains check inside registerToStore, you can avoid doing it here. then you can also avoid the previous line (`val (idSet, store) = store(muxId)`). These 3 lines can just be:
`registerToStore(message.id!!, muxId)`

same for getLocalData

Reply:
Sure, done :)

Reply:
I had initially left it in there to make it clear that we were only registering to the store if the ID is not yet registered to the store. I just put a comment there instead. But maybe we could rename it to `maybeRegisterToStore` instead or something like that.